thrift/types_test.go

```diff
@@ -292,6 +292,10 @@ func TestParseBinary(t *testing.T) {
 			value:  []interface{}{256},
 			errMsg: "failed to parse list of bytes",
 		},
+		{
+			value:  []interface{}{-129},
+			errMsg: "failed to parse list of bytes",
+		},
```

Comment:
Can we add a couple of test cases:
 * values in the < 0 range that are allowed, including the borderline value of 128
 * values in the 128-255 range that are allowed, including the border line value of 255

Ideally, the values in the 128-255 range overlap with the negative values used, so we can show that -1 and 255 are the same in the test